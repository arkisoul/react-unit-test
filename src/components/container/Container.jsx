import React from 'react';

import CounterComponent from '../Counter/Counter';

export default class Container extends React.Component {
    constructor(props) {
        super(props);
        this.state = {clicked: false, count: 0}
        this.handleClick = this.handleClick.bind(this)
        this.handleChange = this.handleChange.bind(this)
    }

    handleClick() {
        this.setState({clicked: true})
    }

    handleChange(e) {
        this.setState({
            [e.target.name]: e.target.value
        })
    }

    render() {
        return (
          <div className="container">
            <input
              type="text"
              name="username"
              onChange={this.handleChange}
              value={this.state.username}
            />
            <input
              type="password"
              name="password"
              onChange={this.handleChange}
              value={this.state.password}
            />
            <button onClick={this.handleClick}>Click me</button>
            <CounterComponent count={this.state.count} />
          </div>
        );
    }
}