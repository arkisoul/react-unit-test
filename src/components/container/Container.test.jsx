import { shallow } from "enzyme";
import Container from "./Container";

describe("test cases for container component", () => {
  const onClickHandler = jest.fn();
  const ContainerComponent = shallow(<Container onClick={onClickHandler} />);
  it("container component is render properly", () => {
    expect(ContainerComponent.getElements()).toMatchSnapshot();
  });
  
  it("container component should have a button element", () => {
    const buttonElement = ContainerComponent.find("button");
    expect(buttonElement.length).toEqual(1);
    expect(buttonElement.text()).toEqual('Click me');
  });

  it('onclick of button should update clicked state from false to true', () => {
    const buttonElement = ContainerComponent.find("button");
    expect(ContainerComponent.state('clicked')).toEqual(false)
    // buttonElement.props().onClick(onClickHandler)
    buttonElement.simulate("click");
    expect(ContainerComponent.state("clicked")).toEqual(true);
    // expect(onClickHandler).toBeCalledTimes(1);
  })

});

describe('container component input elements', () => {
  const ContainerComponent = shallow(<Container />);
  it('should have two input elements', () => {
    const inputs = ContainerComponent.find('input').length
    expect(inputs).toEqual(2)
  })
  
  it('should have one input[type="text"] element', () => {
    const inputs = ContainerComponent.find('input[type="text"]').length
    expect(inputs).toEqual(1)
  })
  
  it('should have one input[type="text"] element handle change event', () => {
    const input = ContainerComponent.find('input[type="text"]')
    input.props().onChange({
      target: { name: "username", value: "johndoe" },
    });
    expect(ContainerComponent.state('username')).toEqual("johndoe");
  })
  
  it('should have one input[type="password"] element', () => {
    const inputs = ContainerComponent.find('input[type="password"]').length
    expect(inputs).toEqual(1)
  })
  
  it('should have one input[type="password"] element handle change event', () => {
    const input = ContainerComponent.find('input[type="password"]')
    input.props().onChange({
      target: { name: "password", value: "12345678" },
    });
    expect(ContainerComponent.state('password')).toEqual('12345678')
  })
})
