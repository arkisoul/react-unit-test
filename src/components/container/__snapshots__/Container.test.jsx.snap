// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test cases for container component container component is render properly 1`] = `
Array [
  <div
    className="container"
  >
    <input
      name="username"
      onChange={[Function]}
      type="text"
    />
    <input
      name="password"
      onChange={[Function]}
      type="password"
    />
    <button
      onClick={[Function]}
    >
      Click me
    </button>
    <Counter
      count={0}
    />
  </div>,
]
`;
